classes/fields.php

```diff
@@ -99,543 +99,699 @@ public static function not_support( $type, $not_support ){
 	 * @return array
 	 */
 	public static function internal_types() {
+		$suffix = defined('SCRIPT_DEBUG') && SCRIPT_DEBUG ? '.min' : '';
+
 		$internal_fields = array(
 			//basic
-			'text'             => array(
-				"field"       => __( 'Single Line Text', 'caldera-forms' ),
-				"description" => __( 'Single Line Text', 'caldera-forms' ),
-				"file"        => CFCORE_PATH . "fields/generic-input.php",
-				"category"    => __( 'Basic', 'caldera-forms' ),
-				"setup"       => array(
-					"template" => CFCORE_PATH . "fields/text/config.php",
-					"preview"  => CFCORE_PATH . "fields/text/preview.php"
-				),
+			'text'               => self::get_text_field(),
+			'hidden'             => self::get_hidden_field(),
+			'email'              => self::get_email_field(),
+			'button'             => self::get_button_field(),
+			'number'             => self::get_number_field(),
+			'phone'              => self::get_phone_field(),
+			'phone_better'       => self::get_phone_field_v2(),
+			'paragraph'          => self::get_paragraph_field(),
+			'wysiwyg'            => self::get_wysiwyg_field(),
+			'url'                => self::get_url_field(),
 
+			//eCommerce
+			'credit_card_number' => self::get_credit_card_number_field(),
+			'credit_card_exp'    => self::get_credit_card_expiration_field(),
+			'credit_card_cvc'    => self::get_credit_card_cvc_field(),
+
+			//special
+			'calculation'        => self::get_calculation_field(),
+			'range_slider'       => self::get_range_slider_field(),
+			'star_rating'        => self::get_star_rating_field(),
+			'utm'                => self::get_utm_field(),
+			'gdpr'               => self::get_gdpr_field(),
+
+			//file
+			'file'               => self::get_file_uploader_field(),
+			'advanced_file'      => self::get_advanced_file_uploader_field(),
+
+			//content
+			'html'               => self::get_html_field(),
+			'summary'            => self::get_summary_field(),
+			'section_break'      => self::get_section_break_field(),
+
+			//select
+			'dropdown'           => self::get_dropdown_field(),
+			'checkbox'           => self::get_checkbox_field(),
+			'radio'              => self::get_radio_field(),
+			'filtered_select2'   => self::get_filtered_select2_field(),
+			'date_picker'        => self::get_date_picker_field(),
+			'toggle_switch'      => self::get_toggle_switch_field(),
+			'color_picker'       => self::get_colorpicker_field(),
+			'states'             => self::get_states_field(),
+
+			//discontinued
+			'recaptcha'          => self::get_recaptcha_field(),
+		);
+
+		return $internal_fields;
+	}
+
+	public static function get_text_field()
```

Comment:
All methods need inline docs: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-functions-class-methods

Reply:
While this may be useful in other context, I think it will be just noise in the code. methods are about 10 lines long and kind of self-documented...

Reply:
I think the only _potential_ useful would be the `@version` comment, but even that i think that adds only very little value.

Reply:
I've added the docblocks too.